docs/lang/articles/misc/ndarray_android.md

```diff
@@ -0,0 +1,449 @@
+				- `shape_offset_in_bytes_in_args_buf`: The offset of the Ndarray's shape information in `args_buf`.
+		- `used.arr_arg_to_bind_idx`: A map specifying the SSBO to bind for a given Ndarray. For example, `"1": 5` (line 48) binds the `vel` Ndarray to the binding index `5`.
+
+*Well, we hope you were not overwhelmed with that much information coming in all at once. In the following section, we will revisit the JSON file, as well as provide tables and graphs that help illustrate some of the concepts and notions listed above.*
+
+### Prepare SSBO and shape information
+
+Before executing the GLES compute shaders in your runtime program, you need to get all your resources ready, including:
+
+- Bind SSBO for the corresponding buffer
+- Bind SSBO for each Ndarray
+- Fill `args_buffer` with Ndarray shape information
+
+#### Bind SSBO for the corresponding buffer
+
+The following table lists the buffers commonly used in a Taichi program together with their binding indexes:
+
+| **Buffer**    | **Global/kernel-spedific** | **Storing**                                                  | **Binding index** |
+| ------------- | -------------------------- | ------------------------------------------------------------ | ----------------- |
+| `root_buffer` | Global                     | All fields with fixed offsets and of fixed sizes.            | `0`               |
+| `gtmp_buffer` | Global                     | Global temporary data                                        | `1`               |
+| `args_buffer` | Kernel-specific            | Arguments passed to the Taichi kernel <ul><li>Scalar arguments</li> <li>Each Ndarray's shape information:  <ul><li>Shape of the Ndarray</li> <li>Element shape</li></ul></li></ul> | `2`               |
+
+1. You *only* need to bind an SSBO for `root_buffer` if your Taichi script uses at least one field.
+2. Bind a small SSBO, say an SSBO of 1,024 Bytes, to `1`, the binding index of `gtmp_buffer`.
+3. Bind an SSBO of 64 x 5 = 320 Bytes to `2`, the binding index of `args_buffer`.
+
+
+
+#### Bind SSBO for each Ndarray
+
+Before running a specific kernel in your runtime program (the `initialize` kernel for example), you must bind SSBO of a proper size for each Ndarray in the kernel in accordance to the value of `used.arr_arg_to_bind_idx`.
+
+The following is a summary of line 29-49 of the above JSON file:
+
+| Ndarray | Taichi kernel | Dimension | Element shape | Argument index | Binding index |
+| ------- | ------------- | --------- | ------------- | -------------- | ------------- |
+| `pos`   | `initialize`  | `1`       | `[2]`         | `0`            | `4`           |
+| `vel`   | `initialize`  | `1`       | `[2]`         | `1`            | `5`           |
+
+If you give each Ndarray a shape [500], and an element shape [2] (meaning that each element is a 2-D vector):
+
+- Each Ndarray has 500 x 2 = 1,000 numbers
+- Because the number type is float, the size of each Ndarray's SSBO is 1,000 x 4 = 4,000 Bytes.
```

Comment:
```suggestion
- Because the number type is float (specified in the python file above), the size of each Ndarray's SSBO is 1,000 x 4 = 4,000 Bytes.
```